scripts/builder/checker_builder/config.py

```diff
@@ -0,0 +1,104 @@
+"""
+Configuration file logic for Checker builds
+"""
+import logging
+import math
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Optional
+
+import yaml
+from jinja2 import Environment, PackageLoader, select_autoescape
+from jinja2.environment import Template
+from marshmallow import Schema, ValidationError, fields
+from marshmallow.decorators import post_load
+
+logging.basicConfig(level=logging.INFO, format="%(message)s")
+logger = logging.getLogger(__name__)
+
+DEFAULT_CONFIG = Path("checker.yaml")
+
+# ================================================================================================
+# Config classes
+# ================================================================================================
+
+# TODO: Additional support for setting certain parameters at build time can be implemented
+# using:
+# def commented(comment: str, default=None) -> Field:
+#     """Creates a dataclass field with a comment in its metadata"""
+#     return field(default=default, metadata={"comment": comment})
+# @dataclass
+# class SystemParameters:
+#     """Checker system parameters"""
+
+#     fminting: Decimal = commented(
+#         "Dimensionless. Factor used for setting the minting limit.",
+#         default=(Decimal(21) / Decimal(10)),
+#     )
+
+
+@dataclass
+class CollateralTokenConfig:
+    scaling_factor: int = 1_000_000
+
+    @property
+    def decimal_digits(self) -> int:
+        return math.floor(math.log10(self.scaling_factor))
+
```

Comment:
Hmmm, it's probably better to go the other way: get the decimal digits as an input, and compute the scaling factor by exponentiation. This way one cannot for example end up with inconsistencies such as
```
scaling factor = 1_234_567
decimal_digits = 6
```
because of `math.floor`. Actually, I think going through python's numbers might be a hazard and precision might vary. What if we used strings instead?

Reply:
Thanks, @gkaracha! This might have been a misunderstanding on my end. Can the scaling factors be set to values other than multiples of ten? If not then I can definitely reverse this.

Reply:
For most (all?) tokens as far as I know scaling factors are powers of 10 (so the name "_**decimal**_ digits" also implies that the scaling factor is a power of 10).

Our fixedpoint is the only exception to this rule I think, since it has a scaling factor of `18446744073709551616` (which is `2^64`), but that's an internal thing; coins as far as I know always use scaling factors that are powers of 10.

Reply:
Okay awesome, thanks for the clarification. I'll swap this around to take the decimal digits as its input.